instrumentation/github.com/labstack/echo/otelecho/config.go

```diff
@@ -39,30 +37,27 @@ func (o optionFunc) apply(c *config) {
 	o(c)
 }
 
-// WithPropagators specifies propagators to use for extracting
-// information from the HTTP requests. If none are specified, global
-// ones will be used.
-func WithPropagators(propagators propagation.TextMapPropagator) Option {
-	return optionFunc(func(cfg *config) {
-		if propagators != nil {
-			cfg.Propagators = propagators
-		}
-	})
+// newConfig creates a new config struct and applies opts to it.
+func newConfig(opts ...Option) *config {
+	c := &config{}
+	for _, opt := range opts {
+		opt.apply(c)
+	}
+
+	return c
 }
 
-// WithTracerProvider specifies a tracer provider to use for creating a tracer.
-// If none is specified, the global provider is used.
-func WithTracerProvider(provider oteltrace.TracerProvider) Option {
-	return optionFunc(func(cfg *config) {
-		if provider != nil {
-			cfg.TracerProvider = provider
-		}
+func WithOTelHTTPOptions(options ...otelhttp.Option) Option {
```

Comment:
this approach was criticized in the issue that I linked before. for example see: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/746#issuecomment-821235264